search/controllers/advanced/__init__.py

```diff
@@ -143,8 +139,13 @@ def search(request_params: MultiDict) -> Response:
                 ) from ex
             except index.OutsideAllowedRange as ex:
                 raise BadRequest(
-                    "Hello clever friend. You can't get results in that range"
-                    " right now."
+                    "You can't get results in that range."
+                ) from ex
+            except Exception as ex:
+                logger.error("Unhandled exception: %s", str(ex))
+                raise InternalServerError(
```

Comment:
Do we want to return an internal server error here?

Reply:
Yes this should return a 500. This is the case where there is an exception that was unexpected and we want to announce that as a 500 so we can notice it and fix it.